github-pullrequest-plugin/src/main/java/com/github/kostyasha/github/integration/branch/events/impl/GitHubBranchCommitEvents.java

```diff
@@ -0,0 +1,147 @@
+        }
+
+        GHCompare.Commit[] commits = getComparedCommits(localBranch, remoteBranch);
+        List<GitHubBranchCause> causes = events.stream()
+                .map(event -> event.check(remoteBranch, localRepo, commits))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+
+        String name = remoteBranch.getName();
+        if (!causes.isEmpty()) {
+            GitHubBranchCause cause = findFilterCause(causes);
+            if (cause != null) {
+                LOGGER.info("Commits for branch [{}] filtered: {}", name, cause.getReason());
+            } else {
+                cause = causes.get(0);
+                LOGGER.info("Commits for branch [{}] allowed: {}", name, cause.getReason());
+            }
+            return cause;
+        }
+
+        return null;
+    }
+
+    // visible for testing to avoid complex mocking
+    GHCompare.Commit[] getComparedCommits(GitHubBranch localBranch, GHBranch remoteBranch) throws IOException {
+        String previous = localBranch.getCommitSha();
+        String current = remoteBranch.getSHA1();
+
+        LOGGER.debug("Comparing previous hash [{}] with current hash [{}]", previous, current);
+        return remoteBranch.getOwner()
+                .getCompare(previous, current)
+                .getCommits();
+    }
+
+    public List<GitHubBranchCommitEvent> getEvents() {
+        return events;
+    }
+
+    public void setEvents(List<GitHubBranchCommitEvent> events) {
+        this.events = events;
+    }
+
+    private GitHubBranchCause findFilterCause(List<GitHubBranchCause> causes) {
+        GitHubBranchCause cause = causes.stream()
+                .filter(GitHubBranchCause::isSkip)
+                .findFirst()
+                .orElse(null);
+
+        if (cause == null) {
+            return null;
+        }
+
+        LOGGER.debug("Cause indicated build should be skipped: {}", cause.getReason());
+        return cause;
+    }
+
+    private GitHubBranchCause toCause(GHBranch remoteBranch, GitHubBranchRepository localRepo, boolean skip, String message,
+            Object... args) {
+        return new GitHubBranchCause(remoteBranch, localRepo, String.format(message, args), skip);
+    }
+
+    @Extension
+    public static class DescriptorImpl extends GitHubBranchEventDescriptor {
+
+        private static final Logger LOG = LoggerFactory.getLogger(GitHubBranchTrigger.class);
+
+        @Override
+        public boolean configure(StaplerRequest req, JSONObject formData) throws FormException {
+            req.bindJSON(this, formData);
+
+            save();
+            return super.configure(req, formData);
+        }
+
+        @Override
+        public final String getDisplayName() {
+            return DISPLAY_NAME;
+        }
+
+        public List<GitHubBranchCommitEventDescriptor> getEventDescriptors() {
+            LOG.error("branch commit events");
```

Comment:
good catch, and it's not. i think i had that in there as an easy way to make sure the code was being called. will remove.